dev/check-delta-connect-codegen-python.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (2024) The Delta Lake Project Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Utility for checking whether generated the Delta Connect Python protobuf codes are out of sync.
+#   usage: ./dev/check-delta-connect-codegen-python.py
+
+import os
+import sys
+import filecmp
+import tempfile
+import subprocess
+
+# Location of your Spark git development area
+DELTA_HOME = os.environ.get("DELTA_HOME", os.getcwd())
```

Comment:
`os.getcwd()` .. using this always breaks when you are running the python from a different location that delta root dir. its brittle. best to use the directory of this file to traverse backwards and find the root dir.

Reply:
see this - https://github.com/delta-io/delta/blob/master/run-tests.py#L220